app/src/main/java/com/otaman/weather/ui/NavGraph.kt

```diff
@@ -0,0 +1,45 @@
+package com.otaman.weather.ui
+
+import androidx.compose.runtime.Composable
+import androidx.navigation.NavGraph.Companion.findStartDestination
+import androidx.navigation.NavHostController
+import androidx.navigation.compose.NavHost
+import androidx.navigation.compose.composable
+import com.otaman.weather.ui.forecast_report_screen.ForecastReportScreen
+import com.otaman.weather.ui.main_screen.MainScreen
+
+@Composable
+fun SetupNavGraph(
+    navController: NavHostController
+) {
+    NavHost(
+        navController = navController,
+        startDestination = Screen.Main.route
+    ) {
+        composable(
+            route = Screen.Main.route
+        ) {
+            MainScreen(onForecastReportButtonClick = {
+                navController.navigateSingleTopTo(Screen.ForecastReport.route)
+            })
+        }
+        composable(
+            route = Screen.ForecastReport.route
+        ) {
+            ForecastReportScreen(onBackClick = {
+                navController.popBackStack()
+            })
+        }
+    }
+}
+
+fun NavHostController.navigateSingleTopTo(route: String) =
+    this.navigate(route) {
+        popUpTo(
+            this@navigateSingleTopTo.graph.findStartDestination().id
+        ) {
+            saveState = true
+        }
+        launchSingleTop = true
+        restoreState = true
+    }
```

Comment:
What's the purpose of this extension?

Reply:
I tried to do this like in codelab. Main purpose was to prevent creating many screens when clicking on button many times. But we can click on button in out app only once so i little edited that.